python/cudf/cudf/utils/cudautils.py

```diff
@@ -776,22 +778,53 @@ def gpu_mark_found_float(arr, val, out, not_found):
             out[i] = not_found
 
 
-def find_first(arr, val):
+@cuda.jit
+def gpu_mark_gt(arr, val, out, not_found):
+    i = cuda.grid(1)
+    if i < arr.size:
+        if check_gt(arr[i], val):
```

Comment:
Was there a reason you needed to turn this into a separate function instead of just doing:
```
if arr[i] > val:
```

Reply:
You're right - these separate functions shouldn't be necessary (afaik).  I will remove them.  If it becomes necessary to cover edge cases the way `check_equals_float` does, seperate functions *can* be added in the future.

Reply:
`check_equals_float` was special cased to handle `NaN` values which are only problematic when we need `NaN` == `NaN` for certain functions.